runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java

```diff
@@ -74,9 +78,15 @@
 	private final JLabel expHour = new JLabel();
 	private final JLabel expLeft = new JLabel();
 	private final JLabel actionsLeft = new JLabel();
+	private final Map<String, Integer> oppInfoHealth = OpponentInfoPlugin.loadNpcHealth();
```

Comment:
I'm pretty sure you're not supposed to call methods from other plugins as they're designed not to rely on each other.

Reply:
Besides that, you're not using oppInfoHealth in this class.

Reply:
I definitely need opponentinfo in the xpstatesingle, is there a way to declare that it has a dependency on that plugin? Can definitely remove this one from here though

Reply:
Under no circumstances should you rely on the internal logic of another plugin, and I'm gradually updating all plugin access modifiers to make this impossible.

If you find yourself logically needing something another plugin already uses internally, then move it out of the plugins to a more logical location within the client and reference it from there.

Reply:
FYI, I'm reworking some of the opponent health info plugin as part of #3939, and will absolutely be making this method private.